org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/presentation/DAPWatchExpression.java

```diff
@@ -32,10 +32,20 @@ public void evaluateExpression(String expression, IDebugElement context, IWatchE
 		if (context.getDebugTarget() instanceof DSPDebugTarget dapDebugger) {
 			final var args = new EvaluateArguments();
 			args.setExpression(expression);
-			DSPStackFrame frame = castNonNull(Adapters.adapt(context, DSPStackFrame.class));
-			args.setFrameId(frame.getFrameId());
-			dapDebugger.getDebugProtocolServer().evaluate(args).thenAccept(
-					res -> listener.watchEvaluationFinished(createWatchResult(dapDebugger, expression, res)));
+
+			@Nullable
+			DSPStackFrame dspStackFrame = Adapters.adapt(context, DSPStackFrame.class);
+			if (dspStackFrame != null) {
+				DSPStackFrame frame = castNonNull(dspStackFrame);
+				args.setFrameId(frame.getFrameId());
+				dapDebugger.getDebugProtocolServer().evaluate(args).thenAccept(
+						res -> listener.watchEvaluationFinished(createWatchResult(dapDebugger, expression, res)));
+			} else {
+				// Program not running at the moment
+				EvaluateResponse res = new EvaluateResponse();
+				res.setResult("(pending)");
+				listener.watchEvaluationFinished(createWatchResult(dapDebugger, expression, res));
```

Comment:
I think this should just set the evaluation to `null` in this case. I am pretty sure that is what is done in other debuggers like  JDT ([link to line 86 of JavaWatchExpressionDelegate](https://github.com/eclipse-jdt/eclipse.jdt.debug/blob/6c5cadc5c3dc4a125d50f566e0e70d5cd9f89c32/org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/JavaWatchExpressionDelegate.java#L86)) when it can't evaluate.

```suggestion
				// Program not running at the moment
				listener.watchEvaluationFinished(null);
```

I think it depends if you want to show an error/warning in this case, or just leave the value field blank.

Reply:
Thank you for the suggestion! I think it makes sense to follow the same approach as JDT, for consistency :-). I have updated my PR to just pass `null` when we can't adapt the context.

Reply:
Just to add - I've tested this version that uses `null`, and it simply leaves the value field blank as one would expect.